add package.yml JSON schema

schnea opened the pull request:
## Summary

Adds a JSON Schema for the package.yml format that makes it possible to receive hints about the format from a LSP. Addresses #104 .

Notes:
- For the possible license values, so far I've grabbed all identifiers on https://spdx.org/licenses/ with a FSF Free/Libre checkmark. But I don't really have a good idea of what that actually means, I feel like the list of licenses you can redistribute is probably a different one? 
- There's a `devel` key mentioned as deprecated in the ypkg docs that I've left out for now.
- Not sure if the source field could be validated some more, the fact that urls are keys makes it a bit tricky

## Test Plan

- add `# yaml-language-server: $schema=/path/to/schema.json` to a package.yml of your choice
- in VSCode, install the YAML Language Support by Red Hat Extension for in-editor validation/auto completion/hover support
- try to create a bogus package.yml and see if it gets flagged.

EbonJaeger commented on data/schema.json:
```diff
@@ -0,0 +1,460 @@
+      "description": "Set to 'true' if this package benefits from being built with Clang."
+    },
+    "conflicts": {
+      "type": "string",
```
This can be either a single string, or an array of strings. Actually, grepping for it in the packages repository, we can also set patterned packages to have conflicts. That's gonna be complicated...

EbonJaeger reviewed (changes requested):
This is awesome! I'm assuming that the schema spec doesn't differentiate between a regular string and a multi-line string (I think they're called scalar strings)?

schnea commented:
> I'm assuming that the schema spec doesn't differentiate between a regular string and a multi-line string (I think they're called scalar strings)?

I don't think there's any differentiation in the spec, they are both json strings. 

> It might be interesting to script this with a schema.json.in template, import the license list from spdx then spit out schema.json as part of the build process. But that is only a suggestion, there may be a nicer way to do it.

I've split out the licenses into their own file now and added a little script to fetch them. Not sure how to add that to the build process though.

EbonJaeger commented:
Is there a directory that the schema is supposed to go in? That can be added to the build system easy enough. As for running the script to generate the licenses list, that we'd have to research; I don't know how to do that.

EbonJaeger reviewed (approved):
LGTM, thanks!
